src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -273,7 +285,32 @@ class MediaBaseViewer extends BaseViewer {
         console.error(err);
 
         const errorCode = getProp(err, 'target.error.code');
-        const errorDetails = errorCode ? { error_code: errorCode } : {};
+        const errorMessage = getProp(err, 'target.error.message');
+        const errorDetails = errorCode ? { error_code: errorCode, error_message: errorMessage } : {};
+
+        // refresh the token if token expired
+        if (errorCode === MediaError.MEDIA_ERR_NETWORK && errorMessage.includes(MEDIA_TOKEN_EXPIRE_ERROR)) {
```

Comment:
What do you think of abstracting this to a common method on `MediaBaseViewer` that can be reused across the different subclasses?

Something like:
```
handleExpiredTokenError(error) {
    if (this.isExpiredTokenError(error)) { ... }
    else {
        this.options.refreshToken().then(this.restartPlayback).catch(error => this.triggerError(error))
    }
}
```

And then the classes that want to use it can implement `isExpiredTokenError` and `restartPlayback`